### ISSUE TITLE
Changed behaviour of composer run-script breaks composer test script conditional

### ISSUE THREAD

**Issue by james-tu2** [heroku/heroku-buildpack-php#559]
Hello!

I noticed in the recent upgrade to Composer v2.3.3 that the behaviour of `composer run-script -l` has been changed.

It currently throws an argument error:

```
❯ composer run-script -l

  Not enough arguments (missing: "script").

run-script [--timeout TIMEOUT] [--dev] [--no-dev] [-l|--list] [--] <script> [<args>...]
```

whereas in Composer v2.2.11 it will list the scripts without any issues:

```
❯ composer run-script -l
scripts:
  post-autoload-dump
  post-root-package-install
  post-create-project-cmd
  test                       Runs the test script as defined in composer.json.
  post-install-cmd
```

I'm not sure if this was intended, as I've opened a discussion to composer asking [here](https://github.com/composer/composer/discussions/10705), but this causes the test script to bypass the `test` script to be bypassed when running buildpack test.

Is there a way for me to pin this to 2.2.11 for Composer here until we resolve the discussion with them?

**Comment by dzuelke** [comment]
Good catch. Reported upstream: https://github.com/composer/composer/issues/10707

Apologies for this regression; our tests didn't cover this because they all use Composer 2.2 for installs; will address this.

In the meantime, I think the quickest way to pin yourself to 2.2 is to `composer self-update --2.2` (available since Composer 2.3.3) and to re-generate `composer.lock`; then Heroku will use 2.2 for your project.

**Comment by dzuelke** [comment]
Another workaround is to specify your `test` script in `app.json`, e.g. like so:

```json
{
  "environments": {
    "test": {
      "scripts": {
        "test": "composer run-script test"
      }
    }
  }
}
```
